### DISCUSSION TITLE
How to access the default font for the Arabic language?

### DISCUSSION THREAD

**Discussion by pitanello** [airsdk/Adobe-Runtime-Support#1302] [Q&A]
Hello
I started using in Adobe Animate the default font "_sans" for all the dialogs in my game, but i have a problem with the Arabic language. Some reviews report that the text is written backwards.
Do you know a way to access Arabic through a default font already installed in the mobile device of the user?
Thanks a lot
Gaetano

**Comment by ajwfrost** [comment]
Hi

You can get support for RTL fonts such as Arabic via use of the "Flash Text Engine". I think Animate used to support this via a "TLF Text Field" component (TLF = Text Layout Framework) but don't think that's included in their latest releases, so you may need to do this manually by placing a text field via ActionScript. You can use the `flash.text.engine.TextBlock` class which can then generate the `TextLine` objects that can be added to the display list; or if you're using Starling/Feathers then these have classes for using FTE rather than the classic TextField.

See https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/text/engine/TextBlock.html

Note that we also have an issue with Android 12 and _sans text that may also impact the Arabic behaviour (#1299)...

thanks

**Reply by pitanello** [reply]
Thanks a lot Andrew, i will give it a try!
